python/ray/rllib/dqn/dqn_evaluator.py

```diff
@@ -112,33 +119,48 @@ def sample(self):
                 obs, actions, rewards, new_obs, dones)
 
         batch = SampleBatch({
-            "obs": obs, "actions": actions, "rewards": rewards,
-            "new_obs": new_obs, "dones": dones,
+            "obs": [pack(o) for o in obs], "actions": actions,
+            "rewards": rewards,
+            "new_obs": [pack(o) for o in new_obs], "dones": dones,
             "weights": np.ones_like(rewards)})
         assert batch.count == self.config["sample_batch_size"]
+
+        # Prioritize on the worker side
+        if self.config["apex_optimizer"]:
+            td_errors = self.dqn_graph.compute_td_error(
+                self.sess, obs, batch["actions"], batch["rewards"],
+                new_obs, batch["dones"], batch["weights"])
+            new_priorities = (
+                np.abs(td_errors) + self.config["prioritized_replay_eps"])
+            batch.data["weights"] = new_priorities
+
         return batch
 
     def compute_gradients(self, samples):
-        _, grad = self.dqn_graph.compute_gradients(
+        td_error, grad = self.dqn_graph.compute_gradients(
             self.sess, samples["obs"], samples["actions"], samples["rewards"],
             samples["new_obs"], samples["dones"], samples["weights"])
-        return grad
+        return grad, td_error
```

Comment:
I don't think we need to actually return this here right?

Reply:
You do if you use another optimizer

Reply:
which optimizer requires this? i thought all other optimizers only return grad

Reply:
Ah I guess theoretically a variant of local_sync_replay that doesn't use the fused compute_apply(). For now, I guess we can remove these two methods entirely.